index.js

```diff
@@ -88,16 +89,17 @@ const baseSchema = {
   },
 }
 
-function concatEnvSchemas(schema, otherSchema) {
-  return {
+function mergeJsonSchemas(schema, otherSchema) {
+  const mergedSchema = {
     type: 'object',
-    required: schema.required.concat(otherSchema.required),
-    properties: {
-      ...schema.properties,
-      ...otherSchema.properties,
-    },
+    properties: mergeDeepRight(schema.properties, otherSchema.properties),
```

Comment:
Can we merge the schemas without introduce ramda dependency? In this case, it should be a first level merge for the properties schema. The spread operator is not enough?

Also for the omit below

Reply:
On that note I just realized that it might be more complicated than what I though. I had to merge the properties because the library env-schema we use to validate and extract the env data, operates only with the top level `properties` field and does not seek the nested `properties` inside the `allOf`.

But, if we keep the properties in a top level, such scenario could appear:

defaultEnv: 
```json
{
  "type": "object",
  "properties": {
    "street_name": { "enum": ["Street", "Avenue", "Boulevard"] }
  }
}
```

serviceEnv: 
```json
{
  "type": "object",
  "properties": {
    "street_name": { "type": "string"}
  }
}
```

In this case, both shallow merge and deep merge would have unwanted results I would say.
I suggest to throw an error when both `defaultEnv` and `serviceEnv` have overlapping properties.
What do you think?

Reply:
Yes, I think it would be ok. It's better to avoid misconfiguration in the more clear possible way